### PR TITLE
std.csv std.stdio.File.byLine example

### PR DESCRIPTION
**Pull request by burner**
IMO it it not obvious how to get the content of a file to the csvReader

the new example shows how to do it

### CODE REVIEW

File: std/csv.d [hunk 1]
```diff
@@ -39,6 +39,18 @@
  *         writefln("%s works as a %s and earns $%d per year",
  *                  record[0], record[1], record[2]);
  *     }
+ *
+ *     // the file with filename.csv would contain the same string as shown
+ *     // above, notice the byLine and joiner
+ *
+ *     auto file = File("filename.csv", "r");
+ *     foreach(record; csvReader!(Tuple!(string,string,int))(
+ *         file.byLine.joiner("\n")))
```
Comment (ZombineDev) [inline std/csv.d]: I think it would look better if you put the `csvReader!(..)` at the end of the UFCS pipeline

Comment (burner) [inline std/csv.d]: done

File: std/csv.d [hunk 2]
```diff
@@ -39,6 +39,18 @@
  *         writefln("%s works as a %s and earns $%d per year",
  *                  record[0], record[1], record[2]);
  *     }
+ *
+ *     // the file with filename.csv would contain the same string as shown
```
Comment (JakobOvrum) [inline std/csv.d]: file with filename.csv? file named filenamed.csv?

File: std/csv.d [hunk 3]
```diff
@@ -39,6 +39,18 @@
...
+ *     // above, notice the byLine and joiner
```
Comment (JakobOvrum) [inline std/csv.d]: this is a run-on sentence, use a semicolon instead of a comma

File: std/csv.d [hunk 4]
```diff
@@ -39,6 +39,18 @@
  *         writefln("%s works as a %s and earns $%d per year",
  *                  record[0], record[1], record[2]);
  *     }
+ *
+ *     // Assume the file filename.csv contains the same string as shown
+ *     // above. The following code shows how to read the data from this file.
```
Comment (WalterBright) [inline std/csv.d]:
```
// To read the same string from the file "filename.csv":
```
Short and sweet!

Comment (burner) [inline std/csv.d]: done

File: std/csv.d [hunk 5]
```diff
@@ -39,6 +39,17 @@
  *         writefln("%s works as a %s and earns $%d per year",
  *                  record[0], record[1], record[2]);
  *     }
+ *
+ *     // To read the same string from the file "filename.csv":
+ *
+ *     auto file = File("filename.csv", "r");
+ *     foreach(record;
+ *         file.byLine.joiner("\n").csvReader!(Tuple!(string,string,int)))
```
Comment (JakobOvrum) [inline std/csv.d]: spaces: `Tuple!(string, string, int)`

Comment (burner) [inline std/csv.d]: fixed

**Comment by JakobOvrum** [comment]
Auto-merge toggled on

**Comment by burner** [comment]
thanks

**Pull request merged**